backend/routers/troops.js

```diff
@@ -18,8 +21,13 @@ async function createTroop(req, res, next) {
   const { _id: owner } = req.user;
   const level = Number.parseInt(req.query.level, 10) || 1;
   try {
+    const cost = parseInt(foxRule.constructionCost, 10);
+    const purchase = await ResourceModel.purchaseItem(owner, cost);
```

Comment:
```suggestion
    const hasEnoughMoney = await ResourceModel.purchaseItem(owner, cost);
```

purchase is not describing what you are checking